features/functests/performance/performance.go

```diff
@@ -50,20 +63,145 @@ var _ = Describe("performance", func() {
+			sysctlMap := map[string]string{
+				"net.core.busy_read":              "50",
+				"net.core.busy_poll":              "50",
+				"net.ipv4.tcp_fastopen":           "3",
+				"kernel.numa_balancing":           "0",
+				"kernel.sched_min_granularity_ns": "10000000",
+				"vm.dirty_ratio":                  "10",
+				"vm.dirty_background_ratio":       "3",
+				"vm.swappiness":                   "10",
+				"kernel.sched_migration_cost_ns":  "5000000",
+			}
+
+			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
+				LabelSelector: perfWorkerNodesLabel,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).Should(BeNumerically(">", 0), "cannot find nodes labeled as "+perfWorkerNodesLabel)
+
+			for _, node := range nodes.Items {
+				mcd, err := mcdForNode(&node)
+				Expect(err).ToNot(HaveOccurred())
+				mcdName := mcd.ObjectMeta.Name
+				for param, expected := range sysctlMap {
+					By(fmt.Sprintf("executing the command \"sysctl -n %s\" inside the pod %s", param, mcdName))
+					Eventually(func() string {
+						out, err := exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace,
+							"-c", perfMachineConfigDaemonContainer, mcdName, "sysctl", "-n", param).CombinedOutput()
+						Expect(err).ToNot(HaveOccurred())
+						return strings.TrimSpace(string(out))
+					}, perfSysctlTimeout*time.Second, perfSysctlPollInterval*time.Second, ).Should(Equal(expected),
+						fmt.Sprintf("parameter %s value is not %s", param, expected))
+				}
+			}
+		})
+	})
+
+	var _ = Context("Tuned kernel parameters", func() {
+		It("Should contain configuration injected through the 12-tuned-worker-rt template", func() {
+			sysctlMap := map[string]string{
+				"kernel.hung_task_timeout_secs": "600",
+				"kernel.nmi_watchdog":           "0",
+				"kernel.sched_rt_runtime_us":    "-1",
+				"vm.stat_interval":              "10",
+				"kernel.timer_migration":        "0",
+			}
+
+			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
+				LabelSelector: perfWorkerNodesLabel,
 			})
 			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).Should(BeNumerically(">", 0), "cannot find nodes labeled as "+perfWorkerNodesLabel)
+
 			for _, node := range nodes.Items {
 				mcd, err := mcdForNode(&node)
 				Expect(err).ToNot(HaveOccurred())
 				mcdName := mcd.ObjectMeta.Name
-				bootLoaderEntries, err := exec.Command("oc", "rsh", "-n", "openshift-machine-config-operator", mcdName,
-					"grep", "-R", "initrd", "/rootfs/boot/loader/entries/").CombinedOutput()
+				for param, expected := range sysctlMap {
+					By(fmt.Sprintf("executing the command \"sysctl -n %s\" inside the pod %s", param, mcdName))
+					Eventually(func() string {
+						out, err := exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace,
+							"-c", perfMachineConfigDaemonContainer, mcdName, "sysctl", "-n", param).CombinedOutput()
+						Expect(err).ToNot(HaveOccurred())
```

Comment:
can you return false here instead of Expect(err).ToNot ?

Reply:
You need to return a string here which does not fulfill the expected string

Reply:
@slintes , i think its better to remove this line (Expect..). In this case we will get more profound output, for example:
    parameter net.core.busy_poll value is not 50
    Expected
        <string>: sysctl: cannot stat /proc/sys/net/core/busy_poll: No such file or directory\ncommand terminated with exit code 255
    to equal
        <string>: 50